framework/BaseClasses/BaseInterface.py

```diff
@@ -41,56 +43,97 @@ def __init__(self):
+      provide a basic reading capability from the xml input file for what is common to all types in the simulation than calls _handleInput
       that needs to be overloaded and used as API. Each type supported by the simulation should have: name (xml attribute), type (xml tag),
       verbosity (xml attribute)
-      @ In, xmlNode, ET.Element, input xml
+      @ In, paramInput, InputParameter, input data from xml
       @ In, variableGroups, dict{str:VariableGroup}, optional, variable groups container
       @ In, globalAttributes, dict{str:object}, optional, global attributes
       @ Out, None
     """
     self.variableGroups = variableGroups if variableGroups is not None else {}
-    if 'name' in xmlNode.attrib.keys():
-      self.name = xmlNode.attrib['name']
+    if 'name' in paramInput.parameterValues:
+      self.name = paramInput.parameterValues['name']
     else:
-      self.raiseAnError(IOError,'not found name for a '+self.__class__.__name__)
-    self.type = xmlNode.tag
-    if globalAttributes is not None:
+      self.raiseAnError(IOError, 'not found name for a '+self.__class__.__name__)
+    self.type = paramInput.getName()
+    if self.globalAttributes is not None:
       self.globalAttributes = globalAttributes
-    if 'verbosity' in xmlNode.attrib.keys() or 'verbosity' in self.globalAttributes:
-      verbGlobal = None if self.globalAttributes is None else self.globalAttributes.get('verbosity')
-      verbLocal = xmlNode.attrib.get('verbosity')
-      self.verbosity = verbLocal if verbLocal is not None else verbGlobal
+    if 'verbosity' in paramInput.parameterValues:
+      self.verbosity = paramInput.parameterValues['verbosity'].lower()
       self.raiseADebug('Set verbosity for '+str(self)+' to '+str(self.verbosity))
-    self._readMoreXML(xmlNode)
+    self._handleInput(paramInput)
     self.raiseADebug('------Reading Completed for:')
     self.printMe()
 
-  def handleInput(self, paramInput, variableGroups=None, globalAttributes=None):
+  def initialize(self, *args, **kwargs):
     """
-      provide a basic reading capability from the xml input file for what is common to all types in the simulation than calls _handleInput
+      Set up this interface for a particular activity
+      @ In, args, list, positional arguments
+      @ In, kwargs, dict, keyword arguments
+    """
+    pass
+
+  def run(self, *args, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, args, list, positional arguments
+      @ In, kwargs, dict, keyword arguments
+    """
+    pass
+
+  ################################
+  # Utility API
+  def provideExpectedMetaKeys(self):
+    """
+      Provides the registered list of metadata keys for this entity.
+      @ In, None
+      @ Out, meta, tuple, (set(str),dict), expected keys (empty if none) and indexes/dimensions corresponding to expected keys
```

Comment:
This line crosses 120 character line length limit.

Reply:
fixed